precompiles/utils/src/xcm.rs

```diff
@@ -0,0 +1,354 @@
+// This file is part of Astar.
+
+// Copyright 2019-2022 PureStake Inc.
```

Comment:
This is completely reused from Moonbeam team?

Reply:
main logic is copied from them, changes are made to make it work with our types.